packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -90,11 +97,25 @@ class MenuContent extends React.Component {
     };
   }
 
-  wrapOnKeyDown(item) {
+  wrapOnKeyDown(item, index) {
     const onKeyDown = item.props.onKeyDown;
     return ((event) => {
       if (event.nativeEvent.keyCode === MenuUtils.KEYCODES.ENTER || event.nativeEvent.keyCode === MenuUtils.KEYCODES.SPACE) {
-        this.props.onRequestNext(item);
+        if (item.props.subMenuItems && item.props.subMenuItems.length > 0) {
+          this.props.onRequestNext(item);
+        }
+      } else if (event.nativeEvent.keyCode === MenuUtils.KEYCODES.RIGHT_ARROW) {
+        if (item.props.subMenuItems && item.props.subMenuItems.length > 0) {
+          this.props.onRequestNext(item);
+        } else {
+          this.props.onRequestClose();
```

Comment:
Is using the right arrow key to close a menu standard?

Reply:
That was for the ability to move between menus in a menu bar... that functionality would probably be better placed in the collapsible menu view and not applicable to a stand alone menu. I'll remove it from here.